### DISCUSSION TITLE
Generated UUIDs are not the same for the same test case across different runs, it indicates that Allure is treating each run as a separate test case instead of updating the existing test case with a consistent UUID. This results in inconsistency of results, having in reports duplicates instead of having test result history

### DISCUSSION THREAD

**Discussion by carpentii** [allure-framework/allure2#2257] [Questions & Support]
I  have setup allure reports for a java testNg framework that runs data driven tests. Each test name and description is taken from an excel file. using data provider method. The issue that i face is that when i'm running test suites multiple times after generating results using mvn allure:serve results are interpreted as new scenarios. I investigated this issue and found out that uuid in json report files for specific scenario is different each time , although data provider file remains the same and test are interpreted by allure as different test case.

**Comment by baev** [comment]
UUIDs are never supposed to be the same. Allure supports the test retries feature. The way it works is that Allure calculates `historyId` for each reported test result. The report shows only the latest result with `historyId`; the rest are hidden (shown on the Retries tab for particular test results). 

Under the hood, `historyId` is a hash of the method's fully classified name & test parameters. So, in order to have no duplicates in the report, you need to make sure all the parameters stay the same across the different runs.

**Reply by bigkingkosta** [reply]
What about cucumber jvm7 concerning historyId? When does the hash of the historyID changed? 
My workaround is now: I change the historyID in the Hooks file in @After for every scenario to the scenario name with version information in its scenario name.
But this works with the additional changes in my Issue ticket (https://github.com/allure-framework/allure-java/issues/1008)
